app/scripts/modules/core/src/function/function.read.service.ts

```diff
@@ -36,7 +36,11 @@ export class FunctionReader {
   ): IPromise<IFunctionSourceData[]> {
     return API.all('functions')
       .withParams({ provider: cloudProvider, functionName: name, region: region, account: account })
-      .get();
+      .get()
+      .then((functions: IFunctionSourceData[]) => {
+        functions = this.functionTransformer.normalizeFunctionSet(functions);
+        return this.$q.all(functions.map(fn => this.normalizeFunction(fn)));
```

Comment:
Can you explain why it is necessary to normalize the function set and then go through and normalize each function? I don't know if this is possible, but can all the normalization happen in one step?

Reply:
The data returned by the `get()` call is of type [{..}].
I did it this way just to be consistent with how an array is handled for normalization. 
Should I use the normalization on the first element in the array ?
Would you like me to do it in some other way ?

Reply:
Got it, no I was just curious your rationale behind splitting up the nomalization.